File: Project/server/src/main/java/it/polito/ai/project/services/database/models/Companion.java
package it.polito.ai.project.services.database.models;

import jdk.nashorn.internal.ir.annotations.Reference;
import lombok.Data;

@Data
public class Companion {
    @Reference
    private User userDetails;

    private PediStop initialStop;
    private PediStop finalStop;
    private CompanionState state;

    public Companion(User userDetails, PediStop initialStop, PediStop finalStop, CompanionState state) {
        this.userDetails = userDetails;
        this.initialStop = initialStop;
        this.finalStop = finalStop;
        this.state = state;
    }

    public Companion() {
    }
}

File: Project/Client/src/app/models/companion.ts
import { Stop } from './stop';
import { UserInfo } from './user';

export enum CompanionState {
    AVAILABLE = "AVAILABLE",
    CHOSEN = "CHOSEN",
    CONFIRMED = "CONFIRMED",
    VALIDATED = "VALIDATED"
}

export class Companion {
    public userDetails: UserInfo;
    public initialStop: Stop;
    public finalStop: Stop;
    public state: CompanionState;
}

